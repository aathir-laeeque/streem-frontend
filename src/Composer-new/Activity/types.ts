import {
  Activity,
  ActivityType,
  Checklist,
  Stage,
  Task,
} from '../checklist.types';

export type ActivityProps = {
  activity: Activity;
  taskId: Task['id'];
};

export type AddNewActivityArgs = {
  activityType: ActivityType;
  checklistId: Checklist['id'];
  orderTree: Activity['orderTree'];
  stageId: Stage['id'];
  taskId: Task['id'];
};

export type DeleteActivityArgs = {
  activityId: Activity['id'];
  taskId: Task['id'];
  stageId: Stage['id'];
};

export enum ActivityErrors {
  E405 = 'PROVIDE_REASON_FOR_YES_NO_ACTIVITY',
  E406 = 'PROVIDE_REASON_FOR_PARAMETER_ACTIVITY_OFF_LIMITS',
  E407 = 'YES_NO_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E408 = 'YES_NO_ACTIVITY_SHOULD_HAVE_EXACTLY_TWO_OPTIONS',
  E409 = 'YES_NO_ACTIVITY_TITLE_CANNOT_BE_EMPTY',
  E410 = 'MULTISELECT_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E411 = 'MULTISELECT_ACTIVITY_OPTIONS_CANNOT_BE_EMPTY',
  E412 = 'SINGLE_SELECT_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E413 = 'SINGLE_SELECT_ACTIVITY_OPTIONS_CANNOT_BE_EMPTY',
  E414 = 'CHECKLIST_ACTIVITY_OPTIONS_NAME_CANNOT_BE_EMPTY',
  E415 = 'CHECKLIST_ACTIVITY_OPTIONS_CANNOT_BE_EMPTY',
  E416 = 'PARAMETER_ACTIVITY_VALUE_CANNOT_BE_EMPTY',
  E417 = 'PARAMETER_ACTIVITY_VALUE_INVALID',
  E418 = 'PARAMETER_ACTIVITY_OPERATOR_CANNOT_BE_EMPTY',
  E419 = 'MATERIAL_ACTIVITY_NAME_CANNOT_BE_EMPTY',
  E420 = 'MATERIAL_ACTIVITY_LIST_CANNOT_BE_EMPTY',
  E421 = 'MATERIAL_ACTIVITY_CANNOT_BE_MANDATORY',
  E422 = 'INSTRUCTION_ACTIVITY_TEXT_CANNOT_BE_EMPTY',
  E423 = 'INSTRUCTION_ACTIVITY_CANNOT_BE_MANDATORY',
}
