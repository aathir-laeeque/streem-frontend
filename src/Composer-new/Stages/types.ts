import { Stage } from '../checklist.types';

export type StageCardProps = {
  index: number;
  isActive: boolean;
  isFirstItem: boolean;
  isLastItem: boolean;
  stage: Stage;
};

export type ReOrderArgs = {
  from: number;
  id: Stage['id'];
  to: number;
};
