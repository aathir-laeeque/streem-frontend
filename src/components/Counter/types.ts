import { decrement, increment, reset } from './actions';

export interface CounterState {
  readonly count: number;
}

export enum CounterActionTypes {
  INCREMENT = '@@test/INCREMENT',
  DECREMENT = '@@test/DECREMENT',
  RESET = '@@test/RESET',
}

export type CounterAction = ReturnType<
  typeof increment | typeof decrement | typeof reset
>;
