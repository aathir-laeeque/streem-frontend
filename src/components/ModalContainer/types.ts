import {
  openModalAction,
  closeModalAction,
  closeAllModalAction,
} from './actions';

export enum ModalNames {
  CREATE_JOB_MODAL = 'CREATE_JOB_MODAL',
  JOB_USER_ASSIGN = 'JOB_USER_ASSIGN',
}

export interface Modal {
  type: string;
  props: Record<string, any>;
}

export interface ModalContainerProps {
  currentModals: Modal[];
  closeModal: () => void;
  closeAllModals: () => void;
}

export enum ModalContainerAction {
  OPEN_MODAL = '@@modal/Container/OPEN_MODAL',
  CLOSE_MODAL = '@@modal/Container/CLOSE_MODAL',
  CLOSE_ALL_MODAL = '@@modal/Container/CLOSE_ALL_MODAL',
}

export interface ModalContainerState {
  readonly currentModals: { type: string; props: Record<string, any> }[];
}

export type ModalContainerActionType = ReturnType<
  typeof openModalAction | typeof closeModalAction | typeof closeAllModalAction
>;
