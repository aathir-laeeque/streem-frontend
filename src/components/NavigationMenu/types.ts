export interface MenuItem {
  icon: React.ReactType;
  name: string;
  path: string;
}
