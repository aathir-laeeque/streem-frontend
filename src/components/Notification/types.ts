import { showNotification } from './actions';

export enum NotificationAction {
  SHOW_NOTIFICATION = '@@notification/SHOW_NOTIFICATION',
}

export type NotificationActionType = ReturnType<typeof showNotification>;

export enum NotificationType {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning',
}
