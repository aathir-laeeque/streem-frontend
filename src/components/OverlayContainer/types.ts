import {
  openOverlayAction,
  closeOverlayAction,
  closeAllOverlayAction,
  updatePropsAction,
} from './actions';

export enum OverlayNames {
  ADD_STOP = 'ADD_STOP',
  ASSIGNED_USER_DETAIL = 'ASSIGNED_USER_DETAIL',
  ASSIGNMENT_SUCCESS = 'ASSIGNMENT_SUCCESS',
  COMPLETE_JOB_WITH_EXCEPTION = 'COMPLETE_JOB_WITH_EXCEPTION',
  COMPLETE_TASK_WITH_EXCEPTION = 'COMPLETE_TASK_WITH_EXCEPTION',
  CONFIRMATION_MODAL = 'CONFIRMATION_MODAL',
  CREATE_JOB_MODAL = 'CREATE_JOB_MODAL',
  JOB_USER_ASSIGN = 'JOB_USER_ASSIGN',
  MEDIA_DETAIL = 'MEDIA_DETAIL',
  SIGNATURE_MODAL = 'SIGNATURE_MODAL',
  SIMPLE_CONFIRMATION_MODAL = 'SIMPLE_CONFIRMATION_MODAL',
  START_JOB_MODAL = 'START_JOB_MODAL',
  SKIP_TASK_MODAL = 'SKIP_TASK_MODAL',
  TASK_ERROR_CORRECTION = 'TASK_ERROR_CORRECTION',
  TASK_MEDIA = 'TASK_MEDIA',
  TASK_USER_ASSIGNMENT = 'TASK_USER_ASSIGNMENT',
  CHECKLIST_REVIEWER_ASSIGNMENT = 'CHECKLIST_REVIEWER_ASSIGNMENT',
  CHECKLIST_REVIEWER_ASSIGNMENT_SUCCESS = 'CHECKLIST_REVIEWER_ASSIGNMENT_SUCCESS',
  SUBMIT_REVIEW_MODAL = 'SUBMIT_REVIEW_MODAL',
  CHECKLIST_REVIEWER_SUBMIT_SUCCESS = 'CHECKLIST_REVIEWER_SUBMIT_SUCCESS',
  CHECKLIST_SENT_TO_AUTHOR_SUCCESS = 'CHECKLIST_SENT_TO_AUTHOR_SUCCESS',
  AUTHORS_DETAIL = 'AUTHORS_DETAIL',
  REVIEWERS_DETAIL = 'REVIEWERS_DETAIL',
  CHECKLIST_REVIEWER_ASSIGNMENT_POPOVER = 'CHECKLIST_REVIEWER_ASSIGNMENT_POPOVER',
  TASK_USERS_ASSIGNMENT = 'TASK_USERS_ASSIGNMENT',
  TIMED_TASK_CONFIG = 'TIMED_TASK_CONFIG',
  INITIATE_SIGNOFF = 'INITIATE_SIGNOFF',
  SIGN_OFF_PROGRESS = 'SIGN_OFF_PROGRESS',
  PASSWORD_INPUT = 'PASSWORD_INPUT',
  SESSION_EXPIRE = 'SESSION_EXPIRE',
  SIGN_OFF_STATE = 'SIGN_OFF_STATE',
  SIGN_COMPLETED_TASKS = 'SIGN_COMPLETED_TASKS',
  SIGNNING_NOT_COMPLETE = 'SIGNNING_NOT_COMPLETE',
  PARAMETER_APPROVAL = 'PARAMETER_APPROVAL',
  EDITING_DISABLED = 'EDITING_DISABLED',
  CHECKLIST_INFO = 'CHECKLIST_INFO',
  SIGN_OFF_INITIATED_SUCCESS = 'SIGN_OFF_INITIATED_SUCCESS',
  SIGN_OFF_SUCCESS = 'SIGN_OFF_SUCCESS',
  RELEASE_SUCCESS = 'RELEASE_SUCCESS',
  SHOW_COMPLETED_JOB_WITH_EXCEPTION_INFO = 'SHOW_COMPLETED_JOB_WITH_EXCEPTION_INFO',
  REVISION_ERROR = 'REVISION_ERROR',
}

export interface CommonOverlayProps<T> {
  type: OverlayNames;
  props?: T;
  key: string;
  closeOverlay: () => void;
  closeAllOverlays: () => void;
  popOverAnchorEl?: Element | ((element: Element) => Element);
}

export interface Overlay {
  type: OverlayNames;
  props?: Record<string, any>;
  popOverAnchorEl?: Element | ((element: Element) => Element);
}

export interface OverlayContainerProps {
  currentOverlays: Overlay[];
  closeOverlay: () => void;
  closeAllOverlays: () => void;
}

export enum OverlayContainerAction {
  OPEN_OVERLAY = '@@overlay/Container/OPEN_OVERLAY',
  CLOSE_OVERLAY = '@@overlay/Container/CLOSE_OVERLAY',
  CLOSE_ALL_OVERLAY = '@@overlay/Container/CLOSE_ALL_OVERLAY',
  UPDATE_PROPS = '@@overlay/Container/UPDATE_PROPS',
}

export interface OverlayContainerState {
  readonly currentOverlays: Overlay[];
}

export type OverlayContainerActionType = ReturnType<
  | typeof openOverlayAction
  | typeof closeOverlayAction
  | typeof closeAllOverlayAction
  | typeof updatePropsAction
>;
