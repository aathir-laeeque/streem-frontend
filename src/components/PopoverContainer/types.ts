import {
  openPopoverAction,
  closePopoverAction,
  closeAllPopoverAction,
} from './actions';

export enum PopoverNames {
  TASK_USER_ASSIGNMENT = 'TASK_USER_ASSIGNMENT',
}

export interface Popover {
  type: string;
  popOverAnchorEl: Element | ((element: Element) => Element);
  props?: Record<string, any>;
}

export interface PopoverContainerProps {
  currentPopovers: Popover[];
  closePopover: () => void;
  closeAllPopovers: () => void;
}

export enum PopoverContainerAction {
  OPEN_POPOVER = '@@popover/Container/OPEN_POPOVER',
  CLOSE_POPOVER = '@@popover/Container/CLOSE_POPOVER',
  CLOSE_ALL_POPOVER = '@@popover/Container/CLOSE_ALL_POPOVER',
}

export interface PopoverContainerState {
  readonly currentPopovers: Popover[];
}

export type PopoverContainerActionType = ReturnType<
  | typeof openPopoverAction
  | typeof closePopoverAction
  | typeof closeAllPopoverAction
>;
