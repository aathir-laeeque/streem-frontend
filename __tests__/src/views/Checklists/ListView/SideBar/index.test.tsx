import { configureStore } from '#store';
import SideBar from '#views/Checklists/ListView/SideBar';
import { SideBarProps } from '#views/Checklists/ListView/SideBar/types';
import React from 'react';
import { Provider } from 'react-redux';
import { create, ReactTestRenderer } from 'react-test-renderer';
import { Store } from 'redux';

import { checklists, jobsProperties } from '#mocks';

describe('<SideBar />', () => {
  let store: Store;
  let snapshot: ReactTestRenderer;
  const props: SideBarProps = {
    sideBarOpen: true,
    selectedChecklist: checklists.data[0],
    closeNav: jest.fn(),
    properties: jobsProperties.data,
  };

  beforeAll(() => {
    store = configureStore({});
    const modal = (
      <Provider store={store}>
        <SideBar {...props} />
      </Provider>
    );
    snapshot = create(modal);
  });

  it('should render and match the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
