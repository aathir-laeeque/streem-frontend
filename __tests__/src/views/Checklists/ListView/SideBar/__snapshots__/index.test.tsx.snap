// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render and match the snapshot 1`] = `
<div>
  <div
    onClick={[MockFunction]}
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <div
    className="sidenav"
    id="sideNav"
  >
    <div
      style={
        Object {
          "margin": 16,
        }
      }
    >
      <button
        className="sc-fznZeY fxACvd sidebar-button"
        onClick={[Function]}
      >
        Start A Job
      </button>
      <button
        className="sc-fzokOt izNgjE sidebar-flat-button"
      >
        Edit Checklist
      </button>
      <button
        className="sc-fzokOt izNgjE sidebar-flat-button"
      >
        View Activity
      </button>
      <div
        className="sidebar-sections"
      >
        <div
          className="sidebar-heading"
        >
          VERSIONING
        </div>
        <div
          className="space-between"
        >
          <div
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <div
              className="sidebar-title"
            >
              Version no.
            </div>
            <div
              className="sidebar-subtitle"
            >
              3
            </div>
          </div>
          <div
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <div
              className="sidebar-title"
            >
              Published on
            </div>
            <div
              className="sidebar-subtitle"
            >
              13 May 2020
            </div>
          </div>
        </div>
        <div
          className="space-between"
        >
          <div
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <div
              className="sidebar-title"
            >
              Created By
            </div>
            <div
              className="sidebar-subtitle"
            >
              Lottie Carroll
            </div>
          </div>
          <div
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <div
              className="sidebar-title"
            >
              Created on
            </div>
            <div
              className="sidebar-subtitle"
            >
              12 March 2020
            </div>
          </div>
        </div>
        <div
          className="sidebar-action"
        >
          View Older Verisons
        </div>
      </div>
      <div
        className="sidebar-sections"
      >
        <div
          className="sidebar-heading"
        >
          REVIEWERS
        </div>
        <div
          className="space-between"
        >
          <div
            className="sidebar-subtitle"
          >
            Delia Gregory
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </div>
        <div
          className="space-between"
        >
          <div
            className="sidebar-subtitle"
          >
            Arthur Elliot
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </div>
        <div
          className="sidebar-action"
        >
          +2 more
        </div>
      </div>
      <div
        className="sidebar-sections"
      >
        <div
          className="sidebar-heading"
        >
          APPROVERS
        </div>
        <div
          className="space-between"
        >
          <div
            className="sidebar-subtitle"
          >
            Delia Gregory
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </div>
        <div
          className="space-between"
        >
          <div
            className="sidebar-subtitle"
          >
            Arthur Elliot
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
            />
          </svg>
        </div>
      </div>
      <div
        className="sidebar-sections"
      >
        <div
          className="sidebar-heading"
        >
          ADDITIONAL ACTIONS
        </div>
        <button
          className="sc-fzokOt izNgjE sidebar-flat-button"
        >
          Print Checklist
        </button>
        <button
          className="sc-fzokOt izNgjE sidebar-flat-button danger"
        >
          Delete Checklist
        </button>
      </div>
    </div>
  </div>
</div>
`;
